src/lib/Client.ts

```diff
@@ -163,12 +167,14 @@ export class Client extends EventEmitter {
   async connect(subscribeData?: SubscriptionOptions) {
     if (this.connectPromise) return this.connectPromise
     return (this.connectPromise = new Promise((resolve, reject) => {
-      const rejectHandler = (err: Error) => {
-        this.connectPromise = null
+      const rejectHandler = (connectionState: ConnectionState) => (err: Error) => {
+        this.connectPromise = null;
+        this.emit(connectionState, err);
         return reject(err)
       }
 
-      this.conn.once('error', rejectHandler)
+      this.conn.once('error', rejectHandler(ConnectionState.Error))
+      this.conn.once('close', rejectHandler(ConnectionState.Closed))
```

Comment:
Do both events always occur? Otherwise we may have a stacked callback leak here

Reply:
close is always called, error is only called, when the connection is closed because of a connection error.

Reply:
Okay, all the events will need be unregistered if an event is thrown - otherwise after a loop of connections and closes, if an error occurs it will trigger the error event multiple times

Reply:
Implemented in commit 48e32fc9850885f2f6cb14bf796c0db4478757a3